File: 2478-longest-nice-subarray/2478-longest-nice-subarray.java
class Solution {
    public int longestNiceSubarray(int[] nums) {
        int n = nums.length, left = 0, maxLength = 0;
        int xorSum = 0, currSum = 0;

        for (int right = 0; right < n; right++) {
            currSum += nums[right];
            xorSum ^= nums[right];

            while (xorSum != currSum) {// bitwise AND is not Zero
                currSum -= nums[left];
                xorSum ^= nums[left];
                left++;
            }

            maxLength = Math.max(maxLength, right - left + 1);

        }

        return maxLength;
    }
}

File: 2478-longest-nice-subarray/2478-longest-nice-subarray.ts
function longestNiceSubarray(nums: number[]): number {
    const n = nums.length;
    let left = 0, bitwiseOr = 0, maxLength = 0;

    for (let right = 0; right < n; right++) {
        while ((bitwiseOr & nums[right]) != 0) {
            bitwiseOr ^= nums[left];  // Remove left element from OR
            left++;
        }

        bitwiseOr |= nums[right];
        maxLength = Math.max(maxLength, right - left + 1);

    }

    return maxLength;
};